cli/prisma2/src/bin.ts

```diff
@@ -91,7 +99,20 @@ async function main(): Promise<number> {
     console.error(result)
     return 1
   }
-  console.log(result)
+
+  const checkResult = await checkpoint.check({
+    product: 'prisma',
+    version: packageJson.version,
+    cache_only: true,
+    disable: isCI,
+  })
+  if (checkResult && checkResult.outdated) {
+    console.error(
+      `\n${chalk.blue('Update available')} ${packageJson.version} -> ${checkResult.current_version}\nRun ${chalk.bold(
```

Comment:
This needs to include the npm tag so alpha users do not accidentally update to `latest` and reverse.

Reply:
Yah, I need to do some more research on this / talk to you or Tim to make sure I code this right.

Reply:
A snippet from one of my old projects using `update-notifier` directly:

```
const updateNotifier = require('update-notifier')
const semver = require('semver')
const chalk = require('chalk')

module.exports = async function (opts) {
  const pkg = require('../../../package.json')
  const distTag = (semver.prerelease(pkg.version) ? semver.prerelease(pkg.version)[0] : 'latest')
  // console.log({distTag})

  const notifier = updateNotifier({
    pkg,
    // updateCheckInterval: 0, // to test use this
    updateCheckInterval: 1000 * 60 * 60 * 24,
    distTag: distTag,
  })
  // console.log('notifier.update', notifier.update)

  if (notifier.update && notifier.update.latest !== pkg.version) {
    // Customize message to also include distTag
    let installCommand = null
    if (distTag === 'latest') {
      installCommand = `npm i -g ${pkg.name}`
    } else {
      installCommand = `npm i -g ${pkg.name}@${distTag}`
    }
    const message = 'Update available ' + chalk.dim(notifier.update.current) + chalk.reset(' → ') + chalk.green(notifier.update.latest) + ' \n' +
      'To update run: ' + chalk.cyan(installCommand)

    // show notification
    notifier.notify({
      message: message,
    })
  }
}
```

The line `const distTag = (semver.prerelease(pkg.version) ? semver.prerelease(pkg.version)[0] : 'latest')` assumes that the version we are using contains the dist-tag. `semver.prerelease()` is defined as:

`prerelease(v): Returns an array of prerelease components, or null if none exist. Example: prerelease('1.2.3-alpha.1') -> ['alpha', 1]`

For our current `2.0.0-preview...` this would fail completely of course (as it is broken) - but I assume we will have to use different configurations before and after release (`prisma2` vs. `prisma`) anyway so this could be accomodated.

Reply:
haha, okay had to add a little hack for `2.0.0-preview` to pick the `latest`.

This is now done in: https://github.com/prisma/checkpoint.prisma.io/commit/c51ea5b24315ba657d3b3b6a177ec5a3eacfdf53